src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java

```diff
@@ -127,15 +136,20 @@ private static void setModules(TestInputConfiguration.Builder testInputConfigBui
         } while (!lines.get(lineNo).startsWith("*/"));
     }
 
-    private static String getFullyQualifiedClassName(String filePath, String moduleName) {
+    private static String getFullyQualifiedClassName(String filePath, String moduleName)
+            throws CheckstyleException {
         String fullyQualifiedClassName;
         if (moduleName.startsWith("com.")) {
             fullyQualifiedClassName = moduleName;
         }
         else {
             final String path = SLASH_PATTERN.matcher(filePath).replaceAll("\\.");
-            final int beginIndex = path.indexOf("com.puppycrawl");
             final int endIndex = path.lastIndexOf(moduleName.toLowerCase(Locale.ROOT));
+            if (endIndex == -1) {
+                throw new CheckstyleException("Unable to resolve module name: " + moduleName
+                + ". Please check for spelling errors or specify fully qualified class name.");
+            }
+            final int beginIndex = path.indexOf("com.puppycrawl");
```

Comment:
Question: Why do we need to check for Checkstyle path?

Reply:
To derive module's fully qualified class name as described in https://github.com/checkstyle/checkstyle/pull/10766#discussion_r699581742

Reply:
Shouldn't the whole line the FQN minus any starting asterisk and spaces. If yes, then we should open an issue to convert this in another PR. If not let me know what I missing.

I bring this up because this will be an issue if the sevntu repo wishes (or is required) to make use of the BDD as mentioned in another PR.

Reply:
This seems to also tie to discussion in https://github.com/checkstyle/checkstyle/pull/10766#discussion_r699606461 .

Reply:
> Shouldn't the whole line be the FQN minus any starting asterisk and spaces.

I didn't understand what you mean here.

Reply:
Ok, I think there are 2 things going on here which is confusing you and was making me misleading.

Path is the file path and not the text from the input file. Our input files have 2 formats now:
Short Class Name:
https://github.com/checkstyle/checkstyle/blob/master/src/test/resources/com/puppycrawl/tools/checkstyle/checks/annotation/annotationlocation/InputAnnotationLocationAnnotation.java#L2
FQ Class Name:
https://github.com/checkstyle/checkstyle/blob/master/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/abstractjavadoc/InputAbstractJavadocCorrectParagraph.java#L2

I was originally thinking why we could get the FQN from those lines, but it is not always the FQN as the short one shows above. So the input file doesn't have all the information to identify the package which is what you need to identify here.

Again, this seems to tie back to https://github.com/checkstyle/checkstyle/pull/10766#discussion_r700193375 and possibly using https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java#L183 to get all the information on the module that we need and don't need to hardcode the specifics.

Reply:
Yes, I'll explore this possibility locally and discuss further if needed.